src/v/hashing/utils.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include <absl/hash/hash.h>
+#include <boost/functional/hash.hpp>
+
+#include <functional>
+
+namespace hash {
+namespace impl {
+/// Pulled from utils/functional.h to prevent making v::utils a dependency of
+/// v::hashing
+template<typename>
+inline constexpr bool always_false_v = false;
```

Comment:
We don't need another instance of this, it should be the same as in `v::utils`, i.e., in the global namespace.